### ISSUE TITLE
Move s-mw.o to the new server

### ISSUE THREAD

**Issue by kghbln** [SemanticMediaWiki/semantic-mediawiki.org#58]
Using s-mw.o does not work yet on the new server.

We are getting:
```
Error: 1267 Illegal mix of collations (latin1_bin,IMPLICIT) and (utf8mb4_general_ci,COERCIBLE) for operation '='
```
Thus we have many query issues on the wiki.

The reason is that MariaDB uses UTF8 all over the place while the database was originally set up on LATIN1.

**Comment by mwjames** [comment]
> The reason is that MariaDB uses UTF8 all over the place while the database
> was originally set up on LATIN1.

I think that would be an unfortunate switch given that we want UTF8 to
work. What about [0, 1, 2, 3]?

[0] https://www.a2hosting.com/kb/developer-corner/mysql/convert-mysql-database-utf-8
[1] https://stackoverflow.com/questions/2543993/convert-a-mysql-database-from-latin-to-utf-8
[2] http://www.alphadevx.com/a/420-Converting-a-MySQL-database-from-latin1-to-utf8
[3] https://docs.moodle.org/31/en/Converting_your_MySQL_database_to_UTF8

**Comment by kghbln** [comment]
> I think that would be an unfortunate switch given that we want UTF8 to work. What about [0, 1, 2, 3]?

I do too. I have converted about 10 database with this issue in the past and I am a bit puzzled why following the established procedure did not work on this wiki. Anyhow, the links offer quite some deviating options which I will try.

I am a bit angry with myself because I knew that the database was corrupted even before doing the move, so ignoring this fact and doing the move without staging was a pretty stupid thing to do.

**Comment by kghbln** [comment]
Let's be frank and face it: the database is corrupted beyond repair. I knew this before but had hoped that it will not be true. 

I have now spent another four hours of time I actually not have to convert the database to utf8. I iterated all methods mentioned in this thread as well as the one I had already in place myself. If working, only three of them are working at all but all I get is: http://www.semantic-mediawiki.net/wiki/Semantic_MediaWiki This is what I had in the first place.

I am out of ideas apart from changing MariaDB to use Latin1.

**Comment by kghbln** [comment]
Anyways I have decided to postpone the move of the wiki until further notice. I will now focus on moving all stuff from server "hades" to the new one since this server will be deleted in a fortnight.

**Comment by mwjames** [comment]
> I have now spent another four hours of time I actually not have to convert
> the database to utf8. I iterated all methods mentioned in this thread as
> well as the one I had already in place myself. If working, only three of
> them are working at all but all I get is:
> http://www.semantic-mediawiki.net/wiki/Semantic_MediaWiki This is what I had
> in the first place.
>
> I am out of ideas apart from changing MariaDB to use Latin1.

Question, could we dump the wiki content, use a blank MariaDB UTF-8
collated DB instance, and import the content without having to deal
with the corrupted DB?

**Comment by kghbln** [comment]
> Question, could we dump the wiki content, use a blank MariaDB UTF-8 collated DB instance, and import the content without having to deal with the corrupted DB?

That's the last resort.

**Comment by JeroenDeDauw** [comment]
Do we lose version history if we do that?

**Comment by kghbln** [comment]
> Do we lose version history if we do that?

No, but we will users and logs, etc. so this is really the last resort.

**Comment by kghbln** [comment]
I guess the best option now is to switch the MariaDB server to a configuration which is working for "www". Since "sandbox" is set up in binary it should work in such an environment, too.

**Comment by mwjames** [comment]
> I guess the best option is not to switch the MariaDB server to a
> configuration which is working for www. Since sandbox is set up in binary it
> should work in such an environment, too.

But then you are facing the same issues in future due to some crappy
latin collation, so, I'd say, make the DB future proof using UTF-8 and
store the logs/user somewhere else as backup in case there are needed.

The setup should be equal (and maintainable) for both boxes so that
any update in future is matter of switching not working on the same
problems all over again.

**Comment by kghbln** [comment]
I believe I have good news for all of us. Since Markus wiki at <korrekt.org> suffered from the same issue however is a bit easier to handle because of the small database size I used this one to poke a bit. In the end I was able to import the database in a sane state with pages like [this](http://korrekt.semantic-mediawiki.net/page/%E8%AF%AD%E4%B9%89Web%E6%8A%80%E6%9C%AF%E5%9F%BA%E7%A1%80) serving as a working grand complication. It is still a rodeo but under controlled circumstances.

The praise goes to [Andy Carter](https://andy-carter.com/blog/what-the-a-exporting-utf-8-data-from-a-latin1-mysql-table).

"www" however was already treated in special ways before as [the current version history (see bottom)](https://www.semantic-mediawiki.org/w/index.php?title=Semantic_MediaWiki&action=history) tells us, so there is still a risk of failure. I just hope that we can continue to work with the current status which is acceptable.

**Comment by kghbln** [comment]
> "www" however was already treated in special ways before as the current version history (see bottom) tells us, so there is still a risk of failure. I just hope that we can continue to work with the current status which is acceptable.

This is utterly disheartening. The path described to migrate does not work. I am getting an armada of mysql errors and in the end the complete page table is lost. Now I will have to work myself through the database manually.

**Comment by kghbln** [comment]
This was a true rodeo, however I believe that I made it. ***Eureka!***

I carefully redacted the database schema before import for it not to already throw errors. In a second step I imported the data and now it appears that everything is fine. As an extra special gift it seems that the issues the old database had are no longer present either, e.g.

- [old  Special:BrokenRedirects](https://www.semantic-mediawiki.org/wiki/Special:BrokenRedirects) vs [new  Special:BrokenRedirects](http://www.semantic-mediawiki.net/wiki/Special:BrokenRedirects)
- [old Main Page history](http://www.semantic-mediawiki.org/w/index.php?title=Semantic_MediaWiki&action=history) vs [new Main Page history](http://www.semantic-mediawiki.net/w/index.php?title=Semantic_MediaWiki&action=history)

@mwjames @JeroenDeDauw Note that I still got warnings from MariaDB during the data import, so I'd appreciate if you perhaps could look around on [the wiki](http://www.semantic-mediawiki.net/wiki/Semantic_MediaWiki) as see if you can find some "spots". @mwjames The file you uploaded last weekend is still to be transferred to the new server, so no worries here.

**Comment by kghbln** [comment]
Omg, nope this is still far away from being perfect. See some of my edits: http://www.semantic-mediawiki.net/wiki/Special:Contributions/Kghbln It appears that the database is referring to arbitrary revisions.

**Comment by kghbln** [comment]
Yep, the wiki does not create new revisions and refers all content to the foaf:knows property.

**Comment by kghbln** [comment]
What I will try is to use the schema of "sandbox" and change the increments to the exported values of "www" on the revision and page tables etc. and see where we go from there.

**Comment by kghbln** [comment]
So I spent another couple of hours on this. One can see from recent changes that we can create pages with a wide variety of different scripts without issues and the problems with the revision table are gone, i.e. the correct revisions are being shown. What we cannot do is pages containing emojies, but I could not care less and this was probably not working before either.

The other issues with the first attempt appear to gone, too, meaning that we now get revisions for every edit, the diffs are correct and the actual content is shown and not some unrelated page.

The problems seen with the broken redirects result from formerly borked links now being displayed "correctly". In the end this is a matter of tidying. The version histories work out, so this is ok too.

- [old  Special:BrokenRedirects](https://www.semantic-mediawiki.org/wiki/Special:BrokenRedirects) vs [new  Special:BrokenRedirects](http://www.semantic-mediawiki.net/wiki/Special:BrokenRedirects)
- [old Main Page history](http://www.semantic-mediawiki.org/w/index.php?title=Semantic_MediaWiki&action=history) vs [new Main Page history](http://www.semantic-mediawiki.net/w/index.php?title=Semantic_MediaWiki&action=history)

The only issue I could find now is that the file namespace is somehow borked for existing files [with multibyte characters in them](http://www.semantic-mediawiki.net/wiki/File:Agenda_Cl%3FmentFlipo.jpg). However these are only a few files which can be fixed manually. I am not sure why some of the PDFs make worries, however they are accessible when clicking on them. Probably related to the PDFHandler extension and me still having to check if the VirtualHost holds the correct rewrite rules. However this does not worry me too much. [Special:ListFiles](http://www.semantic-mediawiki.net/wiki/Special:ListFiles)

Currently a full data rebuild is underway and so far no issues popped up.

@JeroenDeDauw @mwjames If you could have another peep... Perhaps I missed something.

In the end things will not get much better I believe so if you ask me: This is it and we should change over.

Note: I still got warnings from MariaDB upon import, mostly `Warning (Code 1071): Specified key was too long; max key length is 767 bytes` which is due to some tables being myisam before ore `Warning (Code 1366): Incorrect string value: '\x96_the_...' for column 'wl_title' at row 1
Query OK, 1 row affected, 1 warning (0.00 sec)` which is probably what we are seeing on special page "BrokenRedirects".

**Comment by JeroenDeDauw** [comment]
Sounds like we can make the switch to me. I'll go through the various Maps pages once the Maps extension is installed again. Also don't forget the GitHub extension, which is missing on .net.

**Comment by kghbln** [comment]
I successfully ran `refreshLinks.php` which was not able to finish before:

```
Deleting illegal entries from the links tables...
  Checking interval (-INF, INF)
    pagelinks: 0, 66 deleted.
    imagelinks: 0, 2 deleted.
    categorylinks: 0, 8 deleted.
    templatelinks: 0, 9 deleted.
    externallinks: 0, 8 deleted.
    iwlinks: 0 deleted.
    langlinks: 0, 1 deleted.
    redirect: 0 deleted.
    page_props: 0, 70 deleted.
```
Most of the broken redirects are fixed now. Really cool I guess.

**Comment by kghbln** [comment]
> Sounds like we can make the switch to me.

I believe so, too. I will inform Markus tomorrow. Probably James will provide feed-back in case he sees some worry.

> I'll go through the various Maps pages once the Maps extension is installed again. Also don't forget the GitHub extension, which is missing on .net.

They are now on bord again. Note that the Google-API-key is for org. Thus it will not work propery until Markus switched the DNS. s-mw.net will be disabled once this is done.

**Comment by kghbln** [comment]
I just asked Markus to do the DNS change also for "www".

**Comment by kghbln** [comment]
We now also use the new server for this wiki. @mwjames found a couple of issues due to the migration for the myisam, latin1 melange on the old serer to innodb and utf8 on the new server which I mostly worked on. There are some minor things still to do. Thus please do not yet destroy the server which was serving smw.org till today which is not the very old server which will automatically be destroyed next week. @mkroetzsch @JeroenDeDauw FYI

**Comment by kghbln** [comment]
I just cancelled the old root server CX30 (stetsonia) server after doing a last backup. Everything is not on the new cloud server CX41 (sulcorebutia).
